src/rust/engine/options/src/args_tests.rs

```diff
@@ -4,18 +4,26 @@
 use core::fmt::Debug;
 use maplit::hashmap;
 
-use crate::args::Args;
+use crate::args::{Args, ArgsReader};
 use crate::fromfile::test_util::write_fromfile;
+use crate::fromfile::FromfileExpander;
 use crate::{option_id, DictEdit, DictEditAction, Val};
 use crate::{ListEdit, ListEditAction, OptionId, OptionsSource};
 
-fn mk_args<I: IntoIterator<Item = &'static str>>(args: I) -> Args {
-    Args::new(args.into_iter().map(str::to_owned))
+fn mk_args<I>(args: I) -> ArgsReader
+where
+    I: IntoIterator,
+    I::Item: ToString,
```

Comment:
Allows us to accept any iterator of anything that can be turned into an owned string, so that code below can use Vec of `&'static str` as well as Vec of `String` (as created by `format!`).